### PR TITLE
Handling chopped data packages.

### PR DESCRIPTION
**Pull request by BauchBeinePoe**
@GermanBluefox This must be tested.

### CODE REVIEW

File: src/modbus-tcp-client.js [hunk 1]
```diff
@@ -10,6 +10,7 @@ module.exports = stampit()
             currentRequestId    = reqId,
             closedOnPurpose     = false,
             reconnect           = false,
+            buffer              = new Buffer(0),
```
Comment (psorowka) [inline src/modbus-tcp-client.js]: `new Buffer` is officially deprecated in favor of `Buffer.from` or `Buffer.alloc` however this initialization is not needed anyway here, right?

File: src/modbus-tcp-client.js [hunk 2]
```diff
@@ -146,14 +154,14 @@ module.exports = stampit()
...
             reqId += 1;
 
-            var head = Buffer.allocUnsafe(7)
+            var head = Buffer.allocUnsafe(7);
 
-            head.writeUInt16BE(reqId, 0)
-            head.writeUInt16BE(this.protocolVersion, 2)
-            head.writeUInt16BE(pdu.length + 1, 4)
-            head.writeUInt8(this.unitId, 6)
+            head.writeUInt16BE(reqId, 0);
```
Comment (psorowka) [inline src/modbus-tcp-client.js]: ahhh @BauchBeinePoe you prefer semicolon? Sorry I'll try to remember it. we are working with the `standard` linter recommendations (https://www.npmjs.com/package/standard) and dropped semicolons internally. just not used to it.

Comment (BauchBeinePoe) [inline src/modbus-tcp-client.js]: Maybe we can adopt the linter recommendations for this project as well. I see if I can setup my vimrc for that.

File: src/modbus-tcp-client.js (hunk 1 above)
Comment (BauchBeinePoe) [inline src/modbus-tcp-client.js]: Depends on how `Buffer.concat` behaves with undefined as a buffer argument.

Comment (BauchBeinePoe) [inline src/modbus-tcp-client.js]: From [here](https://nodejs.org/api/buffer.html#buffer_class_method_buffer_concat_list_totallength) I would say this is not specified so I would allocate an empty Buffer.

File: src/modbus-tcp-client.js (hunk 2 above)
Comment (psorowka) [inline src/modbus-tcp-client.js]: I am using syntastic for vim that works great with `standard`. I like the minimalist code this encourages and I see more and more projects are migrating, however at the end is a matter of taste.

However I would suggest migrating the project to use strict mode in general 

Another topic would be ES6 features (e.g. `let` and `const` scoping and others) that improve readability and explicitness a lot, however drop compatibility to everything older than Node5 why I understand when open source projects hesitate to use that..

Comment (GermanBluefox) [inline src/modbus-tcp-client.js]: The rest of project is with semicolons. So I think if you make a PR, you should use given style guide.

**Comment by psorowka** [comment]
right sorry I missed that :)

2016-11-22 10:16 GMT+01:00 Stefan Poeter <notifications@github.com>:

File: src/modbus-tcp-client.js (hunk 2 above)
Comment (BauchBeinePoe) [inline src/modbus-tcp-client.js]: I've added a new pull request #86.

**Comment by BauchBeinePoe** [comment]
Wow, @GermanBluefox this test is just so wrong on so many levels that I do not even want to talk about it. Just do it right!!

**Pull request closed without merging**